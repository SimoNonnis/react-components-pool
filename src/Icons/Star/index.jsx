import React from 'react';
import PropTypes from 'prop-types';

/**
 * Star Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function Star({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 100 100"
      width={width}
      height={height}
      className={className}
    >
      <title>Star Icon</title>
      <polygon
        className={classNamePath}
        fill={fill}
        points="100 38.76 62.29 38.11 50 2.45 37.7 38.11 0 38.76 30.11 61.48 19.1 97.54 50 75.94 80.89 97.55 69.88 61.48 100 38.76"
      />
    </svg>
  );
}

Star.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

Star.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'starIcon',
  classNamePath: 'starIconPath',
};

export default Star;
