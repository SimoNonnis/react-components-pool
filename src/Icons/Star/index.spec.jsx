/* eslint-env jest */
import React from 'react';

import Star from './';

describe('Test <Star /> icon', () => {
  it('should render Star icon correctly with default props', () => {
    const wrapper = shallow(<Star />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Star icon correctly with props', () => {
    const wrapper = shallow(<Star fill="#8A2BE2" height={30} width={30} />);
    expect(wrapper).toMatchSnapshot();
  });
});
