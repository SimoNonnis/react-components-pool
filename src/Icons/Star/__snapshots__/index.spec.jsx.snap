// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Star /> icon should render Star icon correctly with default props 1`] = `
<svg
  className="starIcon"
  height={40}
  viewBox="0 0 100 100"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Star Icon
  </title>
  <polygon
    className="starIconPath"
    fill="currentColor"
    points="100 38.76 62.29 38.11 50 2.45 37.7 38.11 0 38.76 30.11 61.48 19.1 97.54 50 75.94 80.89 97.55 69.88 61.48 100 38.76"
  />
</svg>
`;

exports[`Test <Star /> icon should render Star icon correctly with props 1`] = `
<svg
  className="starIcon"
  height={30}
  viewBox="0 0 100 100"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Star Icon
  </title>
  <polygon
    className="starIconPath"
    fill="#8A2BE2"
    points="100 38.76 62.29 38.11 50 2.45 37.7 38.11 0 38.76 30.11 61.48 19.1 97.54 50 75.94 80.89 97.55 69.88 61.48 100 38.76"
  />
</svg>
`;
